File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.cpp
class Solution {
public:
    int numSteps(string& s, int ans = 0, int c = 0) {

        for (auto i = s.size() - 1; i > 0; ans += (s[i--] - '0' + c == 1) ? c = 1, 2 : 1);
        return ans + c;
    }
};

File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution(object):
    def numSteps(self, s):
        """
        :type s: str
        :rtype: int
        """
        ans = 0

        while s != "1":
            if s[-1] == "0":
                s = s[:-1]
            else:
                s = bin(int(s, 2) + 1)[2:]
            ans += 1
        return ans
        